### PR TITLE
mypy improvement

### PR DESCRIPTION
**Pull request by Lee-W**
## Description

This is part of https://github.com/commitizen-tools/commitizen/issues/1053, which intends to catch errors earlier through a more strict mypy setting.

## Checklist

- [x] Add test cases to all the changes you introduce
- [x] Run `./scripts/format` and `./scripts/test` locally to ensure this change passes linter check and test
- [x] Test the changes on the local machine manually
- [x] Update the documentation for the changes

## Additional context

#1053

### CODE REVIEW

File: commitizen/cz/customize/customize.py [hunk 1]
```diff
@@ -68,16 +68,16 @@ def message(self, answers: dict) -> str:
         else:
             return message_template.render(**answers)
 
-    def example(self) -> str | None:
-        return self.custom_settings.get("example")
+    def example(self) -> str:
+        return self.custom_settings.get("example") or ""
```
Comment (woile) [inline commitizen/cz/customize/customize.py:72]: You can set the default for the `.get`, right

**Review by woile — APPROVED** [review]
Lgtm

File: commitizen/cz/customize/customize.py (hunk 1 above)
Comment (Lee-W) [reply commitizen/cz/customize/customize.py:72]: ah, yes, miss this one (also missed the reviewed...). Let me fix it now!

Comment (Lee-W) [reply commitizen/cz/customize/customize.py:72]: mypy doesn't seem to be smart enough to make it right. We probably need to rewrite the setting thing a bit to make it work 🤔 But let's keep it for now

**Pull request merged**